jdaviz/core/template_mixin.py

```diff
@@ -230,32 +276,46 @@ def _update_has_subregions(self):
     def labels(self):
         return [s['label'] for s in self.items if 'label' in s.keys()]
 
+    @cached_property
+    def selected_item(self):
+        for item in self.items:
+            if item['label'] == self.selected:
+                return item
+
     @cached_property
     def selected_obj(self):
-        if self.selected == "Entire Spectrum":
+        if self.selected == self._default_text:
             return None
-        return self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                subset_type="spectral").get(self.selected)
+        subset_type = self.selected_item['type']
+        for viewer_id in self.app.get_viewer_ids():
+            match = self.app.get_subsets_from_viewer(viewer_id,
```

Comment:
Okay, I remember now. I did what I did to aperture photometry because of this:

* #1096